### ISSUE TITLE
Inline Dialog Support

### ISSUE THREAD

**Issue by dsx20** [fgilde/MudBlazor.Extensions#22]
Is there a way to apply the MudBlazor.Extensions functionality for an Inline MudBlazor Dialog?  If not, are there any future plans for adding such support or are the static IDialogService.ShowEx methods the only planned supported technique?

**Comment by dsx20** [comment]
I have since identified a state issue with Inline MudBlazor Dialog visibility (although am unsure if it was the manner in which I was using them, so haven't raised it with that project at this time).  As a result I have changed my solution to use the typical IDialogService.Show/ShowEx calls and subsequently I am now able to easily use this MudBlazor.Extensions library.

To clarify my original intended use.  I was looking to use the inline dialog approach because it allows a declarative razor syntax based solution to defining what otherwise needs to be programmatically defined as DialogParameters.  Generally speaking I try to refrain from programmatically creating EventCallback, RenderFragment, and other such 'complex' Parameter instances outside of razor syntax.

In conclusion, at this stage I no longer require this feature so the issue can be closed with no intention to implement if so desired (until such time that there is further interest).

**Comment by dsx20** [comment]
The saga continues.  A colleague of mine suggested we re-visit this problem given the benefit of using the declarative razor syntax over the programmatically defined C# RenderFragment syntax.  As a solution, for now I have created a wrapper component around a MudDialog which uses reflection to call the private DialogServiceExt.InjectOptionsAsync method during first render (after the dialog reference is available, similar to that which is done within the DialogServiceExt.GetDialogAsync method).  This solution appears to be working well and these dialogs are subsequently included in the DOM based purely on their bound IsVisible property (i.e. no explicit ShowEx / Close calls required).

**Comment by fgilde** [comment]
I never thought about, but it's a cool Idea, and I will try to support Inline Dialogs as well

**Comment by fgilde** [comment]
Ok I done it. Currently I'm refactoring something but it will be available in the next version.
After it's published I write one more time and close issue afterwards

**Comment by fgilde** [comment]
Done in 1.7.41
